atomate/common/powerups.py

```diff
@@ -219,3 +220,54 @@ def set_queue_adapter(
         original_wf.fws[idx_fw].spec["_queueadapter"] = q
 
     return original_wf
+
+
+def powerup_by_kwargs(
+    original_wf: Workflow,
+    add_powerup_module: List[str] = None,

```

Comment:
I don't like this way of specifying additional modules. Can you change this so that:
1. If the powerup name doesn't contain a period then try to import it from the default list of modules ("atomate.vasp.powerups", "atomate.qchem.powerups", "atomate.common.powerups"). If that fails then throw an error saying `"Could not find powerup {}.".format(powerup_name)`
2. If the powerup name contains a period then use everything before the last period as the module path, and everything after the last the period as the powerup name.
3. Make it clear that you can specify a powerup using the full module path in the documentation and that this is the recommended way to use powerups that are not defined in atomate. E.g., `"mypackage.mymodule.powerups.my_powerup"` will use the `my_powerup` powerup imported from `mypackage.mymodule.powerups`.

Reply:
So the way of using the keys as the names is problematic since Mongo does not allow the keys to have dots.
I have slightly reformatted the way the function name and args are passed and added the error message as you have suggested.